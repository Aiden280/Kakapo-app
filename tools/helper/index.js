import fs from 'fs-extra';

// Upload stats.json to `http://webpack.github.io/analyse` for analysis.
// Also you can analyze size with `https://github.com/robertknight/webpack-bundle-size-analyzer`:
// `cat stats.json | analyze-bundle-size`
export const writeStats = stats =>
  fs.writeFile('stats.json', JSON.stringify(stats.toJson(), null, 4));

export default {};
