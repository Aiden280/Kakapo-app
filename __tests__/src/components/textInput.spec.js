import React from 'react';
import { shallow } from 'enzyme';
import TextInput from 'components/ui/textInput/textInput';
import { getData } from '../helper';

function setup(props = {}) {
  const propData = {
    name: 'test',
    placeholder: 'without.translation',
    ...getData('intl'),
    ...props
  };
  return { props, wrapper: shallow(<TextInput {...propData} />) };
}

test('<TextInput/> render', () => {
  const { wrapper } = setup();
  expect(wrapper.type()).toBe('div');
  expect(wrapper.prop('className')).toBe('group');
  expect(wrapper.find('input').length).toBe(1);
  expect(wrapper.find('.highlight').length).toBe(1);
  expect(wrapper.find('.bar').length).toBe(1);
  expect(wrapper.find('label').length).toBe(1);
});

test('<TextInput/> w/o translation', () => {
  const { wrapper } = setup();
  expect(wrapper.find('label').text()).toBe('without.translation');
});

test('<TextInput/> correct value', () => {
  const { wrapper } = setup({ value: 42 });
  expect(wrapper.find('input').props().defaultValue).toBe(42);
});

test('<TextInput/> correct translation', () => {
  const { wrapper } = setup({ placeholder: 'nav.settings' });
  expect(wrapper.find('label').text()).toBe('Settings');
});

test('<TextInput/> loading spinner', () => {
  const { wrapper } = setup({ spinner: true });
  expect(wrapper.find('.spinner').length).toBe(1);
});
