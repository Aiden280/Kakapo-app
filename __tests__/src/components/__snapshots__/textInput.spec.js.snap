// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput/> correct translation 1`] = `
<div
  className="group"
>
  <input
    autoComplete="off"
    defaultValue={undefined}
    name="test"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    required={true}
    type="text"
  />
  <span
    className="highlight"
  />
  <span
    className="bar"
  />
  <label>
    Settings
  </label>
</div>
`;

exports[`<TextInput/> correct value 1`] = `
<div
  className="group"
>
  <input
    autoComplete="off"
    defaultValue={42}
    name="test"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    required={true}
    type="text"
  />
  <span
    className="highlight"
  />
  <span
    className="bar"
  />
  <label>
    without.translation
  </label>
</div>
`;

exports[`<TextInput/> loading spinner 1`] = `
<div
  className="group"
>
  <input
    autoComplete="off"
    defaultValue={undefined}
    name="test"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    required={true}
    type="text"
  />
  <span
    className="highlight"
  />
  <span
    className="bar"
  />
  <label>
    without.translation
  </label>
  <div
    className="input-add-on-item spinner active"
  >
    <div
      className="double-bounce1"
    />
    <div
      className="double-bounce2"
    />
  </div>
</div>
`;

exports[`<TextInput/> render 1`] = `
<div
  className="group"
>
  <input
    autoComplete="off"
    defaultValue={undefined}
    name="test"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    required={true}
    type="text"
  />
  <span
    className="highlight"
  />
  <span
    className="bar"
  />
  <label>
    without.translation
  </label>
</div>
`;

exports[`<TextInput/> w/o translation 1`] = `
<div
  className="group"
>
  <input
    autoComplete="off"
    defaultValue={undefined}
    name="test"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    required={true}
    type="text"
  />
  <span
    className="highlight"
  />
  <span
    className="bar"
  />
  <label>
    without.translation
  </label>
</div>
`;
